### ISSUE TITLE
15C:  GTO i with negative i

### ISSUE THREAD

**Issue by zathras** [zathras/jrpn#110]
From e-mail:

...  But I have a problem with the "GTO i" instruction:
Normally, in order to go to a specific line number (for ex 122), you have to put a negative value (-122) in the register i before the "GTO i", but JRPN-15C produces a "Error 4" in that case.

Is it possible to fix it ?

**Comment by pc1211** [comment]
Please note that while **GTO I** (with negative I value) transfers execution to line number (with that (absolute) value), nothing is said about **GSB I** (with negative value), but it _could_ be implemented in JRPN-15C.

For your information (HP 15c Collector’s Edition Owner's Handbook - Edition 3, February 2023), page 108 & 109):
**Indirect Branching With I**
The I key—but not the (i) key—can be used for indirect branching
(**GTO I**) and subroutine calls (**GSB I**). (Only the integer portion of
the number in Register I is used.) ( **(i)** is only used for indirect addressing of
storage registers.)

**To Labels.** If the Register I value is positive, **GTO I** and **GSB I** will transfer
execution to the label which corresponds to the number in the Index
register (see the above table).
For instance, if the Index register contains 20.00500, then a **GTO I**
instruction will transfer program execution to label A. See the chart
on page 107.
**To Line numbers.** If the RI value is negative, **GTO I** causes branching
to that line number (using the absolute value of the integer portion of the
value in Register I).
For instance, if Register I contains −20.00500, then a **GTO I** instruction will
transfer program execution to program line 020.

**Comment by zathras** [comment]
I just tried this out on my 15C (from 1983), and GTO I with -I works as outlined.  That is to say, the following program gives 78 on the real calculator, and Error 4 on the simulator.  

GSB I is interesting.  When I swap that in, I get this stack:
    X:  8
    Y:  78
    Z:  -7

Single-stepping, the GSB I goes to line 7, as expected, but the RTN instruction takes it to line 8, and not line 6 as one would expect.  So I guess the (undocumented) behavior of GSB I, at least with the original firmware, is to branch to line -I, but set the return position to -I+1.  

This being a simulator of the original 15C, I guess I should be bug-compatible!
```
#  Program produced by JRPN 15C.
#  Generated 2024-7-11 10:13 PDT.
#  Program occupies 14 bytes.

   000 {          } 
   001 { 42 21 11 } f LBL A
   002 {        7 } 7
   003 {       16 } CHS
   004 {    44 25 } STO I
   005 {    22 25 } GTO I
   006 {        6 } 6
   007 {        7 } 7
   008 {        8 } 8
   009 {    43 32 } g RTN

# End.
```

**Comment by pc1211** [comment]
Related to GSB I (I<0): "_set the return position to -I+1_"   or perhaps set the return position to expected return line + 2  ???
The mystery remains: see [https://www.hpmuseum.org/cgi-sys/cgiwrap/hpmuseum/archv020.cgi?read=196480](url)

**Comment by zathras** [comment]
Released in 2.1.15
